presto-bytecode/src/main/java/com/facebook/presto/bytecode/ClassGenerator.java

```diff
@@ -124,6 +129,21 @@ public ClassGenerator dumpClassFilesTo(Optional<Path> dumpClassPath)
         Map<String, byte[]> bytecodes = new LinkedHashMap<>();
 
         for (ClassDefinition classDefinition : classDefinitions) {
+            // We call the simpler class writer first to get any errors out using simpler setting.
+            // This helps when we have large queries that can potentially cause COMPUTE_FRAMES
+            // (used by SmartClassWriter for doing more thorough analysis)
+            ClassWriter simpleClassWriter = new ClassWriter(ClassWriter.COMPUTE_MAXS);
+            classDefinition.visit(simpleClassWriter);
+            try {
+                simpleClassWriter.toByteArray();
+            }
+            catch (ClassTooLargeException | MethodTooLargeException largeCodeException) {
+                throw new PrestoException(GENERATED_BYTECODE_TOO_LARGE, ERROR_LARGE_BYTECODE);
```

Comment:
Can we have this throw some bytecode specific exception and catch that in presto-main so we don't need to introduce dependency to spi in bytecode? presto-bytecode should be a pretty low level independent package.

Reply:
too many places. I guess we need big refactoring :(

Reply:
OK moved it CompilerUtils and handling it there.